src/components/Navbar/AccountSwitcher/index.tsx

```diff
@@ -3,13 +3,63 @@ import clsx from 'clsx';
 
 import styles from './styles.module.scss';
 import { observer } from 'mobx-react-lite';
-import { mdiAccountCircleOutline, mdiAccountSwitch, mdiHomeAccount, mdiShieldAccount } from '@mdi/js';
+import {
+    mdiAccountCircleOutline,
+    mdiAccountSwitch,
+    mdiCircle,
+    mdiHomeAccount,
+    mdiShieldAccount
+} from '@mdi/js';
 import { useStore } from '@tdev-hooks/useStore';
 import Button from '@tdev-components/shared/Button';
 import useIsBrowser from '@docusaurus/useIsBrowser';
 import Popup from 'reactjs-popup';
 import _ from 'lodash';
 import { useLocation } from '@docusaurus/router';
+import Icon from '@mdi/react';
+import User from '@tdev-models/User';
+
+interface SwitchToUserButtonProps {
+    user: User;
+    isInCurrentClass?: boolean;
+}
+
+const SwitchToUserButton = observer(({ user, isInCurrentClass }: SwitchToUserButtonProps) => {
+    const userStore = useStore('userStore');
+    const socketStore = useStore('socketStore');
+
+    return (
+        <div className={clsx(styles.switchToUserButton)}>
+            <Icon
+                path={mdiCircle}
+                size={0.3}
+                color={
+                    (
+                        userStore.isUserSwitched && userStore.viewedUser?.id == user.id
+                            ? (socketStore.connectedClients.get(user.id) || 0) >= 2
```

Comment:
This is for the case where the user is the same as the viewedUser? Is it in this case ever possible that the user is offline?

Reply:
Yes, the idea is to prevent the "viewed user connection" to count as "the user is online". I think it's generally unlikely that you are viewing a user and at the same time need to know their online status, but this keeps it clean IMO.